packages/components/src/font-size-picker/index.native.js

```diff
@@ -0,0 +1,173 @@
+
+function FontSizePicker( {
+	fontSizes = [],
+	disableCustomFontSizes = false,
+	onChange,
+	value: selectedValue,
+} ) {
+	const [ showSubSheet, setShowSubSheet ] = useState( false );
+	const navigation = useNavigation();
+
+	const onChangeValue = ( value ) => {
+		return () => {
+			goBack();
+			onChange( value );
+		};
+	};
+
+	const selectedOption = fontSizes.find(
+		( option ) => option.size === selectedValue
+	) ?? { name: 'Custom' };
+
+	const goBack = () => {
+		setShowSubSheet( false );
+		navigation.goBack();
+	};
+
+	const openSubSheet = () => {
+		navigation.navigate( BottomSheet.SubSheet.screenName );
+		setShowSubSheet( true );
+	};
+	const label = __( 'Font Size' );
+
+	const units = useCustomUnits( {
+		availableUnits: [ 'px', 'em', 'rem' ],
+	} );
+
+	return (
+		<BottomSheet.SubSheet
+			navigationButton={
+				<BottomSheet.Cell
+					label={ label }
+					separatorType="none"
+					value={
+						selectedValue
+							? sprintf(
+									// translators: %1$s: Select control font size name e.g. Small, %2$s: Select control font size e.g. 12px
+									__( '%1$s (%2$s)' ),
+									selectedOption.name,
+									selectedValue
+							  )
+							: __( 'Default' )
+					}
+					onPress={ openSubSheet }
+					accessibilityRole={ 'button' }
+					accessibilityLabel={ selectedOption.name }
+					accessibilityHint={ sprintf(
+						// translators: %s: Select control button label e.g. Small
+						__( 'Navigates to select %s' ),
+						selectedOption.name
+					) }
+				>
+					<Icon icon={ chevronRight }></Icon>
+				</BottomSheet.Cell>
+			}
+			showSheet={ showSubSheet }
+		>
+			<>
+				<BottomSheet.NavigationHeader
+					screen={ label }
+					leftButtonOnPress={ goBack }
+				/>
+				<View style={ styles[ 'components-font-size-picker' ] }>
+					<BottomSheet.Cell
+						customActionButton
+						separatorType="none"
+						label={ __( 'Default' ) }
+						onPress={ onChangeValue( undefined ) }
+						leftAlign={ true }
+						key={ 'default' }
+						accessibilityRole={ 'button' }
+						accessibilityLabel={ __( 'Selected: Default' ) }
+						accessibilityHint={ __(
+							'Double tap to select default font size'
+						) }
+					>
+						<View>
+							{ selectedValue === undefined && (
+								<Icon icon={ check }></Icon>
+							) }
+						</View>
+					</BottomSheet.Cell>
+					{ fontSizes.map( ( item, index ) => {
+						// Only display a choice that we can currenly select.
+						if ( ! parseFloat( item.size ) ) {
+							return null;
+						}
+						return (
+							<BottomSheet.Cell
+								customActionButton
+								separatorType="none"
+								label={ item.name }
+								subLabel={ item.size }
+								onPress={ onChangeValue( item.size ) }
+								leftAlign={ true }
+								key={ index }
+								accessibilityRole={ 'button' }
+								accessibilityLabel={
+									item.size === selectedValue
+										? sprintf(
+												// translators: %s: Select control option value e.g: "Auto, 25%".
```

Comment:
Should we update this? So it has more context with the font size picker.

Reply:
Good idea!